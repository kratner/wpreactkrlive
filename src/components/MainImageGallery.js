import React, { Component } from "react";
import { LinearProgress } from "@material-ui/core";
import { qryPostsFromPrintsCategory } from "./DataAccess/GetDataWPRESTAPI";
// # SCSS

// # CSS
//@import "~react-image-gallery/styles/css/image-gallery.css";

// # Stylesheet with no icons
//node_modules/react-image-gallery/styles/scss/image-gallery-no-icon.scss
//node_modules/react-image-gallery/styles/css/image-gallery-no-icon.css

import ImageGallery from "react-image-gallery";

class MainImageGallery extends Component {
  state = {
    images: []
  };

  componentDidMount() {
    let appUrl = `${qryPostsFromPrintsCategory}`;
    fetch(appUrl)
      .then(data => data.json())
      .then(data => {
        this.setState({
          images: data
        });
      });
  }

  render() {
    let images = this.state.images
      // .filter(node => node.acf.type === "social")
      // .sort((a, b) => (a.acf.weight > b.acf.weight ? 1 : -1))
      .map((node, index) => {
        let title = node.title.rendered;
        let imageURL = node.jetpack_featured_media_url;
        return {
          original: imageURL,
          originalAlt: title,
          originalTitle: title,
          thumbnail: imageURL,
          thumbnailAlt: title,
          thumbnailTitle: title,
          description: title
        };
      });
    return (
      <div className="image-gallery" id="image-gallery-container">
        {images.length === 0 ? (
          <LinearProgress className="progress" />
        ) : (
          <ImageGallery
            items={images}
            thumbnailPosition={this.props.thumbnailPosition}
          />
        )}
      </div>
    );
  }
}

export default MainImageGallery;
