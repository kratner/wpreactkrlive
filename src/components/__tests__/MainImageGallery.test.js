import React from "react";
import { shallow } from "enzyme";
import MainImageGallery from "../MainImageGallery";

describe("MainImageGallery", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<MainImageGallery />);
    expect(wrapper).toMatchSnapshot();
  });
});
