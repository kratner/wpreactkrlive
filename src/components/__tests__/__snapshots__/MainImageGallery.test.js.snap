// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainImageGallery renders correctly 1`] = `ShallowWrapper {}`;
